com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/lightlistbuild.compute

```diff
@@ -414,12 +435,97 @@ void FinePruneLights(uint threadID, int iNrCoarseLights, uint2 viTilLL, float vL
     int l=0;
     // need this outer loop even on xb1 and ps4 since direct lights and
     // reflection lights are kept in separate regions.
+
+#ifdef USE_LIGHTVOLUME_CACHE
+    if (threadID < iNrCoarseLights)
+    {
+        int idxCoarse = GetCoarseLightIndex(threadID, iNrCoarseLights);
+        uint uLightVolume = _LightVolumeData[idxCoarse].lightVolume;
+        StoreLightVolumeCache(threadID, idxCoarse, uLightVolume);
+    }
+
+#if NR_THREADS > PLATFORM_LANE_COUNT
+    GroupMemoryBarrierWithGroupSync();
+#endif
+
+    //When using LDS to cache the volume data, this produces the best most optimal code.
+    //Doing a manual loop like the one below adds an extra cost of .1 ms on ps4 if we use LDS.
+    for (int l = 0; l < iNrCoarseLights; ++l)
```

Comment:
I dont understand why, but the wave compiler  generates much better code when we do the loop naively on top of LDS. 
The biggest win so far has been the uVal extra check in the PIXEL_PER_THREAD, not just the early out, but also the fact that it forces the loop to be dynamic, which is consistent with the loop below.
I tried having the loop below (the old one) wiht the dynamic inner loop + on top of LDS, and it generates slower code. So this is a win of .1 ms, on top of the .2 ms that we get from the dynamic inner loop.

Reply:
Did you check what gets generated for Xbox? FXC can lead to very different results than wave often

Reply:
Checked xbox. Neutral. Deleting the old code.